api/server/modules/generic.js

```diff
@@ -0,0 +1,59 @@
+var date = require('date-and-time');
+var bcrypt = require('bcrypt');
+
+var mongo_ObjectID = require('mongodb').ObjectID;
+
+module.exports = {
+    currentTime: function () {
+        return new Date().getTime();
+    },
+    currentTimestamp: function(){
+        return Math.floor(Date.now() / 1000);
+    },
+    currentDate:function(){
+        var now = new Date();
+        ret = date.format(now, 'YYYY-M-D');  
+        return ret;
+    },
+    currentDateTimeDay:function(){
+        var now = new Date();
+        return date.format(now, 'E YYYY-MMM-DD HH:mm:ss A');
+    },
+    currentIsoDate: function(){
+        return new Date();
+    },
+    encode_password : function( string, callback ){
+        bcrypt.hash( string, 10, function(err, hash){
+            callback( hash );
+        });
+    },
+    match_encode_password : function( p1, p2, callback ){
+        // here 2nd param p2 will be encoded string
+        bcrypt.compare( p1, p2, function(err, val){
+            if (val){
+                callback( true );
+            }else{
+                callback( false );
+            }
+        })
+    },
+    get_random_number : function( ){
+        return Math.floor(Math.random() * 9000) + 1000;
```

Comment:
there must be some node module to generate a random no .. can we u find and use that instead